dadab/dadab.js

```diff
@@ -3,77 +3,101 @@ based on:
  - Simple Image Glitching Processing - datamoshing.com | 2016
  - a pixel sorting sketch from Jerome Martinez | 2017
  */
- 
+
 // output needs to be 1066 x 800
- 
+
+var capture;
+var gif;
 var img;
-var imgFileName = "vera";
-var fileType = "jpg";
+var imgFileName = 'vera';
+var fileType = 'jpg';
+
+var START_FRAME = 15;
+var END_FRAME = 115;
+
+const serverUrl = 'http://localhost:3000';
+
+const socketOptions = {
+  transports: ['websocket'],
+  'force new connection': true
+};
+const socket = io(serverUrl, socketOptions);
 
-function preload() {
+function preload () {
   // load an image from our library to set the canvas createCanvas
-  img = loadImage(imgFileName + "." + fileType);
+  img = loadImage(imgFileName + '.' + fileType); // eslint-disable-line no-undef
 }
 
-
-function setup() {
-  
-  // use only numbers (not variables) for the createCanvas() command, Processing 3
-  var canvas = createCanvas(img.width, img.height);
-  canvas.parent("p5canvas");
+function setup () {
+  var canvas = createCanvas(img.width, img.height); // eslint-disable-line no-undef
+  canvas.parent('p5canvas');
 
   // allows the viewer to recreateCanvas and update surface to image file dimensions
   // surface.setResizable(true);
   // surface.setSize(img.width, img.height);
-  
+
+  // todo: tweak these settings
+  capture = createCapture(VIDEO);
+  capture.size(320, 240);
+  capture.hide();
+  setupGif();
+
   // load image onto surface
-  image(img, 0, 0);
+  // todo: fix the fact that `capture` hides the image
+  image(capture, 0, 0); // eslint-disable-line no-undef
 
   img.loadPixels();
-  noStroke();
- 
-   //set the speed at which rectangles are created; 1 frame = 1 rectangle
-  frameRate(30000);
+  noStroke(); // eslint-disable-line no-undef
+
+  // set the speed at which rectangles are created; 1 frame = 1 rectangle
+  frameRate(30000); // eslint-disable-line no-undef
 }
 
-function draw() {
-  // ellipse(100, 100, 100,100);
-  //
-  var randx = random(0,width);
-  var randy = random(0, height);
-  var randc = img.get(randx, randy);
-  
-  var randx_offset = random(0,8);
-  var randy_offset = random(0,8);
-  
-  /*var randh = random(5,7) + random(3,7);
-  var randw = randh * random(4,8);
-  var rands = random(2,5);*/
-  
-  //
-  var randh = random(2,3) + random(2,6);
-  var randw = randh * random(2,7);
-  var rands = random(2,6);
-  
-  //
-  fill(randc);
-  rect(randx+randx_offset, randy+randy_offset, randw, rands);
-  
-  //apply a posterized filter effect to the colors being randomly generated
+function draw () {
+  var randomX = random(0, width);
+  var randomY = random(0, height);
```

Comment:
I kept getting distracted by seeing `randy`, so I changed the variable names to be more direct.

>Who's Randy?